staging/src/k8s.io/component-base/config/validation/validation.go

```diff
@@ -48,9 +48,6 @@ func ValidateLeaderElectionConfiguration(cc *config.LeaderElectionConfiguration,
 	if cc.LeaseDuration.Duration <= cc.RenewDeadline.Duration {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("leaseDuration"), cc.RenewDeadline, "LeaseDuration must be greater than RenewDeadline"))
 	}
-	if len(cc.ResourceLock) == 0 {
```

Comment:
restore this check, this is generic validation multiple components (even out of tree components) can use